mne/minimum_norm/inverse.py

```diff
@@ -788,26 +818,29 @@ def apply_inverse(evoked, inverse_operator, lambda2=1. / 9.,
     sel = _pick_channels_inverse_operator(evoked.ch_names, inv)
     logger.info('Picked %d channels from the data' % len(sel))
     logger.info('Computing inverse...')
-    K, noise_norm, vertno = _assemble_kernel(inv, label, method, pick_ori)
+    K, noise_norm, vertno, source_nn = _assemble_kernel(inv, label, method,
+                                                        pick_ori)
     sol = np.dot(K, evoked.data[sel])  # apply imaging kernel
 
     is_free_ori = (inverse_operator['source_ori'] ==
-                   FIFF.FIFFV_MNE_FREE_ORI and pick_ori is None)
+                   FIFF.FIFFV_MNE_FREE_ORI and pick_ori != 'normal')
 
-    if is_free_ori:
+    if is_free_ori and pick_ori != 'full':
```

Comment:
You should probably have some check somewhere that a fixed-orientation inverse with `pick_ori='full'` raises an error